data-products/src/main/scala/org/sunbird/analytics/exhaust/OnDemandDruidExhaustJob.scala

```diff
@@ -0,0 +1,256 @@
+      })
+    }
+    val deltaFiles = duplicateDimsDf.saveToBlobStore(storageConfig, format, reportId, Option(Map("header" -> "true")), None)
+    deltaFiles
+  }
+
+  def getDate(pattern: String): SimpleDateFormat = {
+    new SimpleDateFormat(pattern)
+  }
+
+  def processRequest(request: JobRequest, reportConfig:ReportConfig, storageConfig:StorageConfig)(implicit spark: SparkSession, fc: FrameworkContext, sqlContext:SQLContext, sc:SparkContext,config: JobConfig,conf: Configuration): JobRequest = {
+    markRequestAsProcessing(request)
+    val requestBody = JSONUtils.deserialize[RequestBody](request.request_data)
+    val requestParamsBody = requestBody.`params`
+    val reportConfigStr = JSONUtils.serialize(reportConfig)
+
+    var finalConfig = reportConfigStr
+    (requestParamsBody.keys).map(ke => {
+      if(finalConfig.contains(ke))
+      finalConfig = finalConfig.replace("$" + ke, requestParamsBody.get(ke).get.toString)
+    })
+
+    val finalReportConfig = JSONUtils.deserialize[ReportConfig](finalConfig)
+    val druidData : RDD[DruidOutput] = druidAlgorithm(finalReportConfig)
+    val result = CommonUtil.time(druidPostProcess(druidData,request.request_id,JSONUtils.deserialize[ReportConfig](JSONUtils.serialize(finalReportConfig)),storageConfig))
+    val response = result._2;
+    val failedOnDemandDruidRes = response.status.equals("FAILED")
+    if (failedOnDemandDruidRes) {
+      markRequestAsFailed(request, response.statusMsg)
+    } else {
+      if(validateEncryptionRequest(request) == true){
+        val storageConfig = getStorageConfig(config, response.file.head)
+        request.download_urls = Option(response.file);
+        request.execution_time = Option(result._1);
+        processRequestEncryption(storageConfig,request)
+        request.status = "SUCCESS";
+        request.dt_job_completed = Option(System.currentTimeMillis)
+      }
+      else {
+        request.status = "SUCCESS";
+        request.download_urls = Option(response.file);
+        request.execution_time = Option(result._1);
+        request.dt_job_completed = Option(System.currentTimeMillis)
+      }
+    }
+    request
+  }
+  def updateRequestAsync(request: JobRequest)(implicit conf: Configuration, fc: FrameworkContext): CompletableFuture[JobRequest] = {
+
+    CompletableFuture.supplyAsync(new Supplier[JobRequest]() {
+      override def get(): JobRequest = {
+        val res = CommonUtil.time(updateRequest(request))
+        JobLogger.log("Request is zipped", Some(Map("requestId" -> request.request_id, "timeTakenForZip" -> res._1)), INFO)
+        request
+      }
+    })
+  }
+
+  def execute()(implicit spark: SparkSession, fc: FrameworkContext, config: JobConfig, sqlContext:SQLContext, sc:SparkContext, conf: Configuration): Metrics = {
+    val reportConfig = JSONUtils.deserialize[ReportConfig](JSONUtils.serialize(config.modelParams.get("reportConfig")))
+    val jobId : String = reportConfig.id
```

Comment:
The report configurations should be a json map right? We should fetch the report config with the job id.

Reply:
Fixed it